scala-package/core/src/main/scala/org/apache/mxnet/FeedForward.scala

```diff
@@ -230,8 +230,11 @@ class FeedForward private(
       val padded = batch.pad
       val realSize = batchSize - padded
       for ((list, nd) <- outputs zip predExec.outputs) {
-        list += nd.slice(0, realSize).copy()
+        val ndSliced = nd.slice(0, realSize)
+        list += ndSliced.copy()
+        ndSliced.dispose()
```

Comment:
A little concern here as we cannot dispose the whole `nd` but a slice of it. But as tested to be stable, it's better than let customer run it with memory leak. LGTM for now.

Reply:
but what if not `copy()`, simply do `list += nd.slice(0, realSize)`?

Reply:
We tried that and it just breaks for unknown reasons. First inference goes well and second just break.

Reply:
@lanking520 Can you show the exception information?

Reply:
@lanking520 @yzhliu  Is it because the jni doesn't free memory?

Reply:
@liuzx32 The way that Scala works is more like playing around shared pointers. When we create a NDArray, we request a pointer that point to a memory space on the C++ side. If we want to release this piece of memory, we need to call dispose somewhere in the Scala code in order to get it executed. However, if we forgot to do that, Scala/JVM will discard this pointer rather than helping us release the memory and that cause the memory leak.